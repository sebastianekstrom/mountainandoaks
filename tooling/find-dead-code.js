const { exec } = require("child_process");

const findDeadCode = () => {
  exec("ts-prune | wc -l", (error, stdout) => {
    const amount = parseInt(stdout.trim(), 10);

    if (error) {
      console.log(error);
      return process.exit(1);
    }

    if (amount !== 0) {
      const pluralized = amount === 1 ? "instance" : "instances";
      console.log(
        `ðŸš¨ ${amount} ${pluralized} of dead code found. Run 'ts-prune' for more details`
      );
      return process.exit(1);
    }

    console.log("âœ… No instances of dead code found");
  });
};

findDeadCode();
