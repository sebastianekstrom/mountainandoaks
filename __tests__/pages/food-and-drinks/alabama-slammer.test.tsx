import React from "react";
import { render, screen } from "../../../tooling/tests";
import AlabamaSlammer from "pages/food-and-drinks/alabama-slammer";

describe("AlabamaSlammer page", () => {
  it("renders title correctly", () => {
    render(<AlabamaSlammer />);
    const title = screen.queryAllByText(/Alabama Slammer/i);
    expect(title).toBeDefined();
  });

  it("renders the ingress text", () => {
    render(<AlabamaSlammer />);
    const ingress = screen.queryAllByText(
      /Meet the Alabama Slammer, a refreshing cocktail with a sports twist!/i
    );
    expect(ingress).toBeDefined();
  });

  it("renders the ingredients", () => {
    render(<AlabamaSlammer />);
    const ingredient = screen.queryAllByText(/Orange juice/i);
    expect(ingredient).toBeDefined();
  });

  it("renders the method steps", () => {
    render(<AlabamaSlammer />);
    const step = screen.queryAllByText(/Combine all ingredients/i);
    expect(step).toBeDefined();
  });
});
