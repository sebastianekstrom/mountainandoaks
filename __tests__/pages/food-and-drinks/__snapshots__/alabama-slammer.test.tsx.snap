// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlabamaSlammer page renders the correct ingredients 1`] = `
<section
  class="mb-8 lg:mb-0"
  data-testid="INGREDIENT_LIST"
>
  <div>
     
  </div>
  <div
    class="flex flex-col"
  >
    <ul>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Orange juice
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          12
          cl
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Vodka
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          2
          cl
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Southern Comfort
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          2
          cl
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Amaretto
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          2
          cl
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Grenadine
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          1
          cl
        </p>
      </li>
    </ul>
  </div>
</section>
`;
